### PR TITLE
Add SAL annotations to Equals and StrictEquals

### PR DESCRIPTION
**Pull request by agarwal-sandeep**

### CODE REVIEW

File: lib/Runtime/Library/JavascriptProxy.cpp [hunk 1]
```diff
@@ -950,14 +950,16 @@ namespace Js
         if (threadContext->IsDisableImplicitCall())
         {
             threadContext->AddImplicitCallFlags(Js::ImplicitCall_External);
+            *value = FALSE;
```
Comment (Yongqu) [inline lib/Runtime/Library/JavascriptProxy.cpp]: nice to have, but no actual bugs here: after adding the explicitcallflag the other data shouldn't be used.

**Comment by Yongqu** [comment]
>         *value = false;

FALSE as well, given we are here? :)


---
Refers to: lib/Runtime/Library/GlobalObject.cpp:2242 in 45535dd. [](commit_id = 45535dd562285d15e5613d3accd2588562a7b1b2, deletion_comment = False)

File: lib/Runtime/Types/RecyclableObject.h [hunk 2]
```diff
@@ -278,8 +278,8 @@ namespace Js {
         virtual BOOL GetEnumerator(BOOL enumNonEnumerable, Var* enumerator, ScriptContext * requestContext, bool preferSnapshotSemantics = true, bool enumSymbols = false);
         virtual BOOL ToPrimitive(JavascriptHint hint, Var* value, ScriptContext * requestContext);
         virtual BOOL SetAccessors(PropertyId propertyId, Var getter, Var setter, PropertyOperationFlags flags = PropertyOperation_None);
-        virtual BOOL Equals(Var other, BOOL* value, ScriptContext* requestContext);
-        virtual BOOL StrictEquals(Var other, BOOL* value, ScriptContext* requestContext);
+        virtual BOOL Equals(__in Var other, __out BOOL* value, ScriptContext* requestContext);
+        virtual BOOL StrictEquals(__in Var other, __out BOOL* value, ScriptContext* requestContext);
```
Comment (Yongqu) [inline lib/Runtime/Types/RecyclableObject.h]: does this require the output to be set on return (compiler warning otherwise?)

**Comment by Yongqu** [comment]
:shipit:

File: lib/Runtime/Types/RecyclableObject.h (hunk 2 above)
Comment (agarwal-sandeep) [inline lib/Runtime/Types/RecyclableObject.h]: Code analysis shows C6101 warning if out parameter is not set in some paths

---
In reply to: [73197400](https://github.com/Microsoft/ChakraCore/pull/1365#discussion_r73197400) [](ancestors = 73197400)

**Comment by agarwal-sandeep** [comment]
@dotnet-bot build arm please

**Comment by agarwal-sandeep** [comment]
@dotnet-bot test Windows arm_debug please

**Comment by agarwal-sandeep** [comment]
@dotnet-bot test Windows arm_test please

**Pull request merged**